rewriter/GenCallAsm.cpp

```diff
@@ -280,21 +280,309 @@ static std::string sig_string(const CAbiSignature &sig,
   return ss.str();
 }
 
+static void call_string(
+    const std::optional<std::string> target_name,
+    WrapperKind kind,
+    AsmWriter &aw,
+    Arch arch) {
+
+  add_comment_line(aw, "Call wrapped function");
+
+  if (arch == Arch::X86) {
+    if (!target_name) {
```

Comment:
This check and the following assert are duplicated. Couldn't you pull them out of the arch checks to avoid this?

Reply:
I reversed the order of these nested ifs to deduplicate the asserts.